src/engine/strat_engine/serde_structs.rs

```diff
@@ -11,24 +11,20 @@
 // structs. These contain simple, serde-friendly data types, and we
 // can convert to or from them when saving our current state, or
 // restoring state from saved metadata.
-//
-// Look for "to_save" and "setup" methods, that either return or take
-// the below structs as parameters.
 
-use std::collections::HashMap;
-use std::path::PathBuf;
+use std::marker::Sized;
 
-use devicemapper::Sectors;
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct BlockDevSave {
-    pub devnode: PathBuf,
-    pub total_size: Sectors,
+/// Implements saving and restoring from metadata.
+pub trait DSerializable<T> {
+    fn to_save(&self) -> T;
+    fn setup(T) -> Self
+        where Self: Sized
+    {
+        unimplemented!()
+    }
```

Comment:
1. ok cool thanks. Maybe add some version of this as a comment in the source?
2. `Saveable` or `StratSerializable`? `Persistable`? `Summarized` doesn't seem accurate since the info being saved is not shortened or simplified, which is what summary connotes to me. 🚲 🏠